src/OrchardCore.Modules/Orchard.ContentFields/Drivers/NumericFieldDriver.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Globalization;
+using System.Threading.Tasks;
+using Orchard.ContentFields.Settings;
+using Orchard.ContentFields.ViewModels;
+using Orchard.ContentManagement;
+using Orchard.ContentManagement.Display.ContentDisplay;
+using Orchard.ContentManagement.Display.Models;
+using Orchard.ContentManagement.Metadata.Models;
+using Orchard.DisplayManagement.ModelBinding;
+using Orchard.DisplayManagement.Views;
+using Microsoft.Extensions.Localization;
+
+namespace Orchard.ContentFields.Fields
+{
+    public class NumericFieldDisplayDriver : ContentFieldDisplayDriver<NumericField>
+    {
+        private readonly CultureInfo _cultureInfo;
+
+        public NumericFieldDisplayDriver(IStringLocalizer<LinkFieldDisplayDriver> localizer)
+        {
+            T = localizer;
+
+            _cultureInfo = CultureInfo.InvariantCulture; // Todo: Get CurrentCulture from site settings
+        }
+
+        public IStringLocalizer T { get; set; }
+
+        public override IDisplayResult Display(NumericField field, BuildFieldDisplayContext context)
+        {
+            return Shape<DisplayNumericFieldViewModel>("NumericField", model =>
+            {
+                model.Field = field;
+                model.Part = context.ContentPart;
+                model.PartFieldDefinition = context.PartFieldDefinition;
+            })
+            .Location("Content")
+            .Location("SummaryAdmin", "");
+        }
+
+        public override IDisplayResult Edit(NumericField field, BuildFieldEditorContext context)
+        {
+            return Shape<EditNumericFieldViewModel>("NumericField_Edit", model =>
+            {
+                var settings = context.PartFieldDefinition.Settings.ToObject<NumericFieldSettings>();
+                model.Value = (field.IsNew() && field.Value == null) ? settings.DefaultValue : Convert.ToString(field.Value, _cultureInfo);
```

Comment:
I am thinking we should maybe provide the New/Update boolean in the EditorContext, we will often have this case, either in fields and parts. Same thing in the Update method.